numba/np/arraymath.py

```diff
@@ -791,6 +792,7 @@ def flat_all(a):
 @overload_method(types.Array, "any")
 def np_any(a):
     def flat_any(a):
```

Comment:
```suggestion
def np_any(a):
    if not type_can_asarray(a):
        raise errors.TypingError('The argument "a" must be array-like')
    def flat_any(a):
```
Suggest catching any type which cannot be converted to an array at typing time, and also testing this.